server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -433,7 +434,8 @@ private void executeRequest(
             requestOperationsListeners = searchRequestOperationsCompositeListenerFactory.buildCompositeListener(
                 originalSearchRequest,
                 logger,
-                TraceableSearchRequestOperationsListener.create(tracer, requestSpan)
+                TraceableSearchRequestOperationsListener.create(tracer, requestSpan),
+                new SearchTaskRequestOperationsListener(taskResourceTrackingService)
```

Comment:
Do you need to create new `SearchTaskRequestOperationsListener` for every request? Should it be singleton instance?

Reply:
+1, IIRC there's only 1 instance of the `TaskResourceTrackingService` per node and it looks like this is doing something very similar.

Reply:
Made `SearchTaskRequestOperationsListener` as singleton in the latest